src/components/loader/loader.test.tsx

```diff
@@ -7,12 +7,26 @@ import Logger from "../../__internal__/utils/logger";
 
 jest.mock("../../hooks/useMediaQuery", () => ({
   __esModule: true,
-  default: jest.fn().mockReturnValue(false),
+  default: jest.fn(),
 }));
 
+const mockUseMediaQuery = useMediaQuery as jest.MockedFunction<
+  typeof useMediaQuery
+>;
+
+beforeEach(() => {
+  jest.clearAllMocks();
+  mockUseMediaQuery.mockReturnValue(true);
+});
+
+afterAll(() => {
+  jest.restoreAllMocks();
+});
+
 testStyledSystemMargin(
   (props) => <Loader data-role="loader" {...props} />,
-  () => screen.getByTestId("loader"),
+  // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+  () => screen.queryByTestId("loader")!,
```

Comment:
suggestion(non-blocking): this could stay as getByTestID("loader")

That would mean we don't have to lint ignore anymore